posthog/async_migrations/setup.py

```diff
@@ -119,3 +123,39 @@ def get_async_migration_dependency(migration_name: str) -> Optional[str]:
         return None
 
     return ASYNC_MIGRATION_TO_DEPENDENCY[migration_name]
+
+
+def verify_required_migrations_have_been_completed(ignore_posthog_version: bool, applied_migrations: set):
+    unapplied_migrations = set(ALL_ASYNC_MIGRATIONS.keys()) - applied_migrations
+    for migration_name, migration in ALL_ASYNC_MIGRATIONS.items():
+        if (
+            (not ignore_posthog_version)
+            and (migration_name in unapplied_migrations)
+            and (POSTHOG_VERSION > Version(migration.posthog_max_version))
+        ):
+            raise ImproperlyConfigured(f"Migration {migration_name} is required for PostHog versions above {VERSION}.")
+
+
+def is_fresh_install(first_migration: str):
+    # To verify we're on a fresh install
+    # (1) check that the first async migration is not started
+    # (2) check that there are no persons exist (i.e. someone didn't try to upgrade to 1.40 from before async migrations introduction)
```

Comment:
```suggestion
    # (2) check that no persons exist (i.e. someone didn't try to upgrade to 1.40 from before async migrations introduction)
```

a) does this check hold true today? (what if I use PostHog purely for feature flags? even more relevant now with the restructure)
b) will this check always hold true in the future

Reply:
Sorry for the accidental commit @tiina303 - GitHub induced me to click new buttons

Reply:
These checks are done as OR, I should clarify the comment. So if (1) or (2). For all versions above 1.34 we'll have the first migration completed as it's one of the ones required before upgrading to 1.34. However we need (2) because folks might be trying to upgrade from before async migrations were introduced to 1.40 directly (not reading major upgrade notes), to make sure we then don't mark all migrations as completed we need (2) to opt out here. These old versions of PostHog everyone had persons at least.